UnitTests/OIDServiceDiscoveryTests.m

```diff
@@ -398,6 +398,12 @@ - (void)testSecureCoding {
   OIDServiceDiscovery *unarchived = [NSKeyedUnarchiver unarchiveObjectWithData:data];
 
   XCTAssertEqualObjects(discovery.discoveryDictionary, unarchived.discoveryDictionary, @"");
+  
+  if (@available(iOS 11.0, *)) {
+    data = [NSKeyedArchiver archivedDataWithRootObject:discovery requiringSecureCoding:YES error:&error];
```

Comment:
This breaks the build for older versions of other platforms where the `NSKeyedArchiver`/`NSKeyedUnarchiver` are not available.

Need more platform checks:
```
if (@available(iOS 11, macCatalyst 13, macOS 10.13, tvOS 11, *)) {
  ...
}
```